test/e2e/framework/util.go

```diff
@@ -5187,18 +5188,19 @@ func WaitForPersistentVolumeClaimDeleted(c clientset.Interface, ns string, pvcNa
 	return fmt.Errorf("PersistentVolumeClaim %s is not removed from the system within %v", pvcName, timeout)
 }
 
-func GetClusterZones(c clientset.Interface) (sets.String, error) {
-	nodes, err := c.CoreV1().Nodes().List(metav1.ListOptions{})
-	if err != nil {
-		return nil, fmt.Errorf("Error getting nodes while attempting to list cluster zones: %v", err)
-	}
-
+func GetClusterZones(nodes *v1.NodeList) (sets.String, error) {
+	notfoundPartNodes := []string{}
 	// collect values of zone label from all nodes
 	zones := sets.NewString()
 	for _, node := range nodes.Items {
 		if zone, found := node.Labels[kubeletapis.LabelZoneFailureDomain]; found {
 			zones.Insert(zone)
+		} else {
+			notfoundPartNodes = append(notfoundPartNodes, node.Name)
 		}
 	}
+	if len(notfoundPartNodes) != 0 {
```

Comment:
If the error is returned above, this `len` check is no longer necessary

Reply:
Done